plugins/web/opentelemetry-plugin-document-load/src/meterDocumentLoad.ts

```diff
@@ -0,0 +1,101 @@
+import { BaseMetricPlugin } from '@opentelemetry/core/build/src/platform/browser';
+import { otperformance } from '@opentelemetry/core';
+import { MetricPluginConfig } from '@opentelemetry/api';
+import { MetricObservable } from '@opentelemetry/metrics';
+import {
+    hasKey,
+    PerformanceEntries,
+    PerformanceLegacy,
+    PerformanceTimingNames as PTN,
+  } from '@opentelemetry/web';
+
+
+export class meterDocumentLoad extends BaseMetricPlugin<unknown> {
+    constructor(config: MetricPluginConfig = {}) {
+        super('@opentelemetry/plugin-document-load');
+        this._onDocumentLoaded = this._onDocumentLoaded.bind(this);
+        this._config = config;
+    }
+
+    private _onDocumentLoaded() {
+        // Timeout is needed as load event doesn't have yet the performance metrics for loadEnd.
+        // Support for event "loadend" is very limited and cannot be used
+        window.setTimeout(() => {
+            this._collectPerformance();
+        });
+    }
+
+    private _collectPerformance() {
+        const entries = this._getEntries();
+        console.log(entries);
+        const loadTimeObserver = this._meter.createObserver('document-load', {
+            description: 'Documet load time',
+        });
+        const loadTime = new MetricObservable();
+        loadTimeObserver.setCallback((observerResult) => {
+            observerResult.observe(loadTime, {});
+        });
+        loadTime.next(Number(entries[PTN.LOAD_EVENT_END]) - Number(entries[PTN.FETCH_START]));
+    }
+
+    private _getEntries() {
+    const entries: PerformanceEntries = {};
+    const performanceNavigationTiming = (otperformance.getEntriesByType(
+      'navigation'
+    )[0] as unknown) as PerformanceEntries;
+
+    if (performanceNavigationTiming) {
+      const keys = Object.values(PTN);
+      keys.forEach((key: string) => {
+        if (hasKey(performanceNavigationTiming, key)) {
+          const value = performanceNavigationTiming[key];
+          if (typeof value === 'number' && value > 0) {
+            entries[key] = value;
+          }
+        }
+      });
+    } else {
+      // // fallback to previous version
+      const perf: typeof otperformance & PerformanceLegacy = otperformance;
+      const performanceTiming = perf.timing;
+      if (performanceTiming) {
+        const keys = Object.values(PTN);
+        keys.forEach((key: string) => {
+          if (hasKey(performanceTiming, key)) {
+            const value = performanceTiming[key];
+            if (typeof value === 'number' && value > 0) {
+              entries[key] = value;
+            }
+          }
+        });
+      }
+    }
```

Comment:
A lot of this logic is essentially the same. I would create a helper function to do so, then you could focus on ensuring that your logic on which PerformanceEntries object you're using is the correct one.

Reply:
As noted [here](https://github.com/tinylox/opentelemetry-js/pull/4#discussion_r441806375), consider changing keys.forEach to a for loop to be consistent with yourself.